BLE_BatteryLevel/source/main.cpp

```diff
@@ -106,14 +106,12 @@ class BatteryDemo : ble::Gap::EventHandler {
     }
 
     void update_sensor_value() {
-        if (_ble.gap().getState().connected) {
-            _battery_level++;
-            if (_battery_level > 100) {
-                _battery_level = 20;
-            }
-
-            _battery_service.updateBatteryLevel(_battery_level);
+        _battery_level++;
+        if (_battery_level > 100) {
+            _battery_level = 20;
         }
+
+        _battery_service.updateBatteryLevel(_battery_level);
```

Comment:
Do we want to update the characteristic if the device isn't connected ?

Reply:
@pan-: Well, `_ble.gap().getState().connected` is deprecated without any replacements (as far as I'm aware of). So now we queue this function in on connection and remove it on disconnection.

Reply:
I thought `onDisconnectionComplete` will be called and no longer call this function. Isn't that what will happen?

Reply:
@hugueskamba I think so, and the way it is looks good to me